ssl/ssl_locl.h

```diff
@@ -1127,6 +1131,9 @@ struct ssl_st {
         int psk_kex_mode;
     } ext;
 
+    /* Parsed form of the ClientHello, kept around across early_cb calls. */
+    struct clienthello_msg_st *clienthello;
```

Comment:
IIRC, the typedef is later in the file, so we'd have to reorder things in order to use it here.